ads/inabox/inabox-messaging-host.js

```diff
@@ -224,15 +224,26 @@ export class InaboxMessagingHost {
       return this.iframeMap_[sentinel];
     }
 
-    // Walk up on the window tree until find host's direct child window
-    while (source.parent !== this.win_ && source !== this.win_.top) {
-      source = source.parent;
-    }
-
-    // Find the iframe element that hosts the message source window
+    // Verify that the source of the postMessage corresponds to one of
+    // the iframes that was registered.
     for (let i = 0; i < this.iframes_.length; i++) {
       const iframe = this.iframes_[i];
       if (iframe.contentWindow === source) {
+	  // Walk up on the window tree and verify that there is no cross-domain
+	  // frame in between the source of the postMessage and the host doc.
+	  // If there is, the host doc will not be able to accurately measure
+	  // the creative's positions in the page, so return null.
+	  // Limit to 10 iterations.
+	  let tempWin = source.parent;
+	  for (let i = 0; i < 10; i++) {
+	      if (tempWin == this.win_ || tempWin == this.win_.top) {
+		  break;
+	      }
+	      if (!canInspectWindow(tempWin)) {
+		  return null;
```

Comment:
let's dev().warn here preferably with actionable info though I suppose we expect it in the passback case though we are indicating in the ad request that inabox is not eligible.  I still wonder what we should do about this case as we need to ensure we meet the same level of viewability measurement support.

Reply:
done